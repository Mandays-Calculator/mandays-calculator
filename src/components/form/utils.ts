import type { FormErrors } from "./types";
import { defaultTo, get } from "lodash";

export const getOption = <Type extends object>(
  path: keyof Type | "value" | "label",
): ((option: Type) => string) => {
  return (option: Type) => {
    return defaultTo(get(option, path), "");
  };
};

export const getFieldError = (
  errors: FormErrors | undefined,
  name: string,
): string | undefined => {
  if (errors) {
    return typeof get(errors, name) === "string" ? get(errors, name) : "";
  }
};
