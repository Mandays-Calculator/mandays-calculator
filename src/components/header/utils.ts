import { UserProject } from "~/api/auth";

export const getProjectOptions = (projects: UserProject[]): SelectObject[] => {
  return projects
    .filter((project: UserProject) => project.active)
    .map((project) => ({
      label: project.name,
      value: project.projectId,
    }));
};
