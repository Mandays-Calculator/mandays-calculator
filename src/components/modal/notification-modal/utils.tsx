import { ReactElement } from "react";
import { ModalType } from "./types";

import Grid from "@mui/material/Grid";
import WarningIcon from "@mui/icons-material/Warning";
import Typography, { TypographyProps } from "@mui/material/Typography";

import { SvgIcon } from "~/components";

interface IdleTimerSectionProps {
  idleTimer: number;
  idleTimerSeconds: number;
}

const StyledTypography: React.FC<TypographyProps> = (
  props: TypographyProps
): ReactElement => (
  <Typography
    variant="inherit"
    fontSize="3.5rem"
    fontFamily="sans-serif"
    {...props}
  />
);

export const IdleTimerSection: React.FC<IdleTimerSectionProps> = ({
  idleTimer,
  idleTimerSeconds,
}: IdleTimerSectionProps): ReactElement => (
  <Grid container spacing={1}>
    <Grid item xs={1} marginX="7rem">
      <StyledTypography>{idleTimer}min</StyledTypography>
    </Grid>
    <Grid item>
      <StyledTypography>{idleTimerSeconds}secs</StyledTypography>
    </Grid>
  </Grid>
);

export const renderIcon = (type: ModalType): ReactElement => {
  switch (type) {
    case "error":
    case "warning":
    case "unauthorized":
      return <SvgIcon name="warning" $size={8} />;
    case "idleTimeOut":
      return <WarningIcon color="error" sx={{ fontSize: 40 }} />;
    case "success":
    default:
      return <WarningIcon color="success" />;
  }
};

export const TimerSection = ({
  idleTimer,
  idleTimerSeconds,
}: {
  idleTimer: number;
  idleTimerSeconds: number;
}): ReactElement => (
  <>
    <IdleTimerSection
      idleTimer={idleTimer}
      idleTimerSeconds={idleTimerSeconds}
    />
    <Grid container spacing={3}>
      <Grid item xs={12}>
        <Typography variant="h5" style={{ margin: "1rem" }}>
          Please click "Stay Logged On" to keep working; or click "Logout" to
          end your session now.
        </Typography>
      </Grid>
    </Grid>
  </>
);
