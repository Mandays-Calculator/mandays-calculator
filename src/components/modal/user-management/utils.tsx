export const genders = [
  {
    label: "FEMALE",
    value: "FEMALE",
  },
  {
    label: "MALE",
    value: "MALE",
  },
  {
    label: "NON-BINARY",
    value: "NON_BINARY",
  },
  {
    label: "PREFER NOT TO SAY",
    value: "PREFER_NOT_TO_SAY",
  },
];

export const rolesData = [
  {
    label: "System Admin",
    value: "ROLE_SYS_ADMIN",
  },
  {
    label: "Sprint Manager",
    value: "ROLE_SPRINT_MANAGER",
  },
  {
    label: "User",
    value: "ROLE_USER",
  },
];
