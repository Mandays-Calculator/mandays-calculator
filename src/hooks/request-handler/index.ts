export * from "./useRequestHandler";
export * from "./types";
