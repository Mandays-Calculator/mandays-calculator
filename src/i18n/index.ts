import i18n from "i18next";
import { initReactI18next } from "react-i18next";
import Backend from "i18next-http-backend";
import LanguageDetector from "i18next-browser-languagedetector";

import translations from "./locales";

const i18nConfig = {
  resources: translations,
  fallbackLng: "en",
  debug: true,
  defaultNS: "translations",
  interpolation: {
    escapeValue: false, // not needed for react as it escapes by default
  },
};

i18n.use(Backend).use(LanguageDetector).use(initReactI18next).init(i18nConfig);

export default i18n;
