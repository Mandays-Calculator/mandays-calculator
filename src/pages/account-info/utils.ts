import { object, string } from "yup";

export const actInfoChangePassSchema = () =>
  object().shape({
    currentPassword: string().required(),
    newPassword: string().required(),
    confirmNewPassword: string().required(),
  });
