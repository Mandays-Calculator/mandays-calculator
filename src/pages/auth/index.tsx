export { default as Auth } from "./Auth";
export { default as ChangePassword } from "./ChangePassword";
export { default as ForgotPassword } from "./ForgotPassword";
export { default as Login } from "./Login";
export * from "./routes";
