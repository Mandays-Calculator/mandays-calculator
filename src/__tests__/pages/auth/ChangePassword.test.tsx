import { QueryClient, QueryClientProvider } from 'react-query';
import { BrowserRouter, useSearchParams } from 'react-router-dom';
import { I18nextProvider } from 'react-i18next';

import { render, waitFor } from '@testing-library/react';

import { Formik } from 'formik';
import i18n from '~/i18n';

import { useResetPasswordMutation } from '~/mutations/auth';
import { useRequestHandler } from '~/hooks/request-handler';
import { ChangePassword } from '~/pages/auth';

import { cleanAllCallback } from './utils/auth-utils';
import userEvent from '@testing-library/user-event';

// Mock the i18next translation function
i18n.init({
    resources: {
        en: {
            translation: {
                "changePassword": {
                    "label": {
                        "createNewPassword": "Create New Password",
                        "enterNewPassword": "Enter new password",
                        "confirmNewPassword": "Confirm new password"
                    },
                    "btnlabel": {
                        "changePassword": "Change Password",
                        "cancel": "Cancel"
                    },
                    "validationInfo": {
                        "charCount": "Password must be at least 8 characters long",
                        "uppercase": "Password must contain an uppercase letter",
                        "lowercase": "Password must contain a lowercase letter",
                        "number": "Password must contain a number",
                        "symbol": "Password must contain one of the following symbols (#$-_!)",
                        "match": "New password and confirm new password must match."
                    }
                },
            },
        },
    },
    lng: 'en',
    interpolation: { escapeValue: false }, // react already safes from xss
});

jest.mock("react-router-dom", () => ({
    ...jest.requireActual('react-router-dom'),
    useSearchParams: jest.fn(),
}));
jest.mock("~/mutations/auth");
jest.mock("~/hooks/request-handler");

const mockUseSearchParams = useSearchParams as jest.MockedFunction<
    typeof useSearchParams
>;
const mockUseResetPasswordMutation = useResetPasswordMutation as jest.MockedFunction<
    typeof useResetPasswordMutation
>;
const mockUseRequestHandler = useRequestHandler as jest.MockedFunction<
    typeof useRequestHandler
>;

afterEach((done) => {
    cleanAllCallback(done);
});

afterAll((done) => {
    cleanAllCallback(done);
});

const queryClient = new QueryClient();
const handleSubmit = jest.fn();

describe('GIVEN ChangePassword component is called,', () => {
    test('WHEN user changes password, THEN it should display correctly', async () => {
        const { container, getByRole } = renderChangePassword(false, false);
        const user = userEvent.setup();

        await user.click(getByRole('button', { name: 'changePassword.btnlabel.changePassword' }));
        await user.click(getByRole('button', { name: 'changePassword.btnlabel.cancel' }));

        await waitFor(() => {
            expect(container).toMatchSnapshot(`change-password`);
        });
    });

    test('WHEN user succesfully changes the password, THEN it should display a success message', async () => {
        const { container, getByRole } = renderChangePassword(true, true);
        const user = userEvent.setup();

        await user.click(getByRole('button', { name: 'login.btnlabel.signIn' }));

        await waitFor(() => {
            expect(container).toMatchSnapshot(`change-password-success`);
        });
    });
});

const renderChangePassword = (codeParam: boolean, successChangePassword: boolean) => {
    runChangePasswordMocks(codeParam, successChangePassword);

    return render(
        <QueryClientProvider client={queryClient}>
            <BrowserRouter>
                <I18nextProvider i18n={i18n}>
                    <Formik initialValues={{}} onSubmit={handleSubmit}>
                        <ChangePassword />
                    </Formik>
                </I18nextProvider>
            </BrowserRouter>
        </QueryClientProvider>
    );
};

const runChangePasswordMocks = (codeParam: boolean, successChangePassword: boolean): void => {
    const useSearchParamsData = [
        {
            get: jest.fn(() => codeParam)
        }
    ];

    mockUseSearchParams.mockImplementation(() => useSearchParamsData as unknown as ReturnType<
        typeof useSearchParams
    >);

    const useResetPasswordMutationData = {
        mutate: null
    }

    mockUseResetPasswordMutation.mockImplementation(() => useResetPasswordMutationData as unknown as ReturnType<
        typeof useResetPasswordMutation
    >);

    const useRequestHandlerData = [
        {
            loading: false,
            error: {
                message: '',
                errorCode: '',
            },
            success: successChangePassword,
        },
        jest.fn()
    ];

    mockUseRequestHandler.mockImplementation(() => useRequestHandlerData as unknown as ReturnType<
        typeof useRequestHandler
    >);
};
