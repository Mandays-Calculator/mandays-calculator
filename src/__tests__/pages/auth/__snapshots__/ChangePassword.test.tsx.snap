// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN ChangePassword component is called, WHEN user changes password, THEN it should display correctly: change-password 1`] = `
<div>
  <form
    data-testid="form-component"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-1gw5rvg-MuiTypography-root"
    >
      changePassword.label.createNewPassword
    </p>
    <div
      class="MuiGrid-root MuiGrid-container css-14b6npa-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-yp0tso-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1exsy04-MuiTypography-root"
        >
          changePassword.label.enterNewPassword
        </p>
        <div
          class="MuiStack-root css-hwnj0i-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="text-field-component"
            field="[object Object]"
            form="[object Object]"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r0:"
                placeholder="changePassword.placeholder.password"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="icon-button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1exsy04-MuiTypography-root"
        >
          changePassword.label.confirmNewPassword
        </p>
        <div
          class="MuiStack-root css-hwnj0i-MuiStack-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
            data-testid="text-field-component"
            field="[object Object]"
            form="[object Object]"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="true"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                id=":r1:"
                placeholder="changePassword.placeholder.confirmPassword"
                type="password"
                value=""
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                  data-testid="icon-button"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="VisibilityOffIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-ihdtdm"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1hzi63y-MuiGrid-root"
      >
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="red-icon-password-length"
              focusable="false"
              style="color: red;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          changePassword.validationInfo.charCount
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="red-icon-password-uppecase"
              focusable="false"
              style="color: red;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          changePassword.validationInfo.uppercase
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="red-icon-password-lowercase"
              focusable="false"
              style="color: red;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          changePassword.validationInfo.lowercase
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="red-icon-password-number"
              focusable="false"
              style="color: red;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          changePassword.validationInfo.number
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="red-icon-password-symbol"
              focusable="false"
              style="color: red;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          changePassword.validationInfo.symbol
        </li>
        <li
          class="MuiListItem-root MuiListItem-gutters css-1nxmd3h-MuiListItem-root"
        >
          <div
            class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="red-icon-password-match"
              focusable="false"
              style="color: red;"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </div>
          changePassword.validationInfo.match
        </li>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-v8fy29-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth css-lthxue-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          changePassword.btnlabel.cancel
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          >
            <span
              class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
              style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
            >
              <span
                class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
              />
            </span>
          </span>
        </button>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-v8fy29-MuiGrid-root"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-fullWidth css-las5yg-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          changePassword.btnlabel.changePassword
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;

exports[`GIVEN ChangePassword component is called, WHEN user succesfully changes the password, THEN it should display a success message: change-password-success 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-5pv5pv-MuiGrid-root"
    >
      <div
        class="css-1yh4p1c"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-9vsojy-MuiSvgIcon-root"
          data-testid="CheckIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9 16.17 4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
          />
        </svg>
        <h6
          class="MuiTypography-root MuiTypography-h6 css-1f7553v-MuiTypography-root"
        >
          changePassword.label.title
        </h6>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-5pv5pv-MuiGrid-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-1oetjkv-MuiTypography-root"
      >
        changePassword.label.success
      </p>
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-pf30pc-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      login.btnlabel.signIn
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      >
        <span
          class="css-y4cjyz-MuiTouchRipple-ripple MuiTouchRipple-ripple MuiTouchRipple-rippleVisible"
          style="width: 2.8284271247461903px; height: 2.8284271247461903px; top: -1.4142135623730951px; left: -1.4142135623730951px;"
        >
          <span
            class="MuiTouchRipple-child MuiTouchRipple-childLeaving"
          />
        </span>
      </span>
    </button>
  </div>
</div>
`;
